my_enumerable.rb

```diff
@@ -0,0 +1,17 @@
+module MyEnumerable
+  def all?
+    each { |num| return false unless yield num }
+    true
+  end
+
+  def any?
+    each { |num| return true if yield num }
+    false
+  end
+
+  def filter
+    filtered = []
+    each { |num| filtered.push(num) if yield num }
+    filtered
+  end
```

Comment:
- [ ] `Filter` method returns an array containing all elements of enum for which the given block returns a true value. let us say if 
`list = MyList.new(1, 2, 3, 4)`
`list.filter {|e| e.even?}`
`[2, 4]`